csharp/src/Ice/LocatorInfo.cs

```diff
@@ -686,39 +670,27 @@ private void
             }
         }
 
-        private readonly ILocatorPrx _locator;
         private ILocatorRegistryPrx? _locatorRegistry;
         private readonly LocatorTable _table;
         private readonly bool _background;
 
-        private Dictionary<string, Request> _adapterRequests = new Dictionary<string, Request>();
-        private Dictionary<Identity, Request> _objectRequests = new Dictionary<Identity, Request>();
+        private readonly Dictionary<string, Request> _adapterRequests = new Dictionary<string, Request>();
+        private readonly Dictionary<Identity, Request> _objectRequests = new Dictionary<Identity, Request>();
     }
 
-    public sealed class LocatorManager
+    public sealed partial class Communicator
```

Comment:
I would move Communicator to the top of the file. It's unclear if the other classes in that file need to be public. (In a future refactoring)